src/shared/FeatureFlagStore.ts

```diff
@@ -0,0 +1,40 @@
+import { observable } from 'mobx';
+import { parseUrl } from 'query-string';
+import _ from 'lodash';
+
+const FEATURE_FLAG_ID = 'featureFlags';
+
+export class FeatureFlagStore {
+    @observable.ref currentState: string[] = [];
+
+    constructor() {
+        const url = parseUrl(window.location.href);
+
+        let storageState =
+            localStorage.getItem(FEATURE_FLAG_ID)?.split(',') || [];
+
+        let newState = url.query.featureFlags?.toString().split(',') || [];
+
+        this.currentState = _.uniq(storageState.concat(newState));
+
+        localStorage.setItem(FEATURE_FLAG_ID, this.currentState.join(','));
+    }
+
+    clear() {
+        localStorage.removeItem(FEATURE_FLAG_ID);
+        this.currentState = [];
+    }
+
+    remove(flag: string) {
+        let storageState =
+            localStorage.getItem(FEATURE_FLAG_ID)?.split(',') || [];
+        let newState = _.reject(storageState, s => s === flag);
+        this.currentState = newState;
+        localStorage.setItem(FEATURE_FLAG_ID, this.currentState.join(','));
+
+        window.location.href = window.location.href.replace(
+            /featureFlags=[^&]+/,
+            ''
+        );
+    }
```

Comment:
Maybe we can add another method `contains(flag: string)` or `has(flag: string)`, because eventually, we want to know if I can enable a feature or not, so we should have a function for that purpose. What do you think?